crates/iceberg-catalog/src/catalog/tables.rs

```diff
@@ -993,16 +920,96 @@ impl<C: Catalog, A: AuthZHandler, S: SecretStore>
     }
 }
 
-pub(crate) fn require_no_location_specified(location: &Option<String>) -> Result<()> {
-    if location.is_some() {
-        return Err(ErrorModel::builder()
-            .code(StatusCode::BAD_REQUEST.into())
-            .message("Specifying a Table `location` is not supported. Location is managed by the Catalog.".to_string())
-            .r#type("LocationNotSupported".to_string())
-            .build()
+fn determine_table_ident(
+    parameters_ident: TableIdent,
+    request_ident: Option<TableIdent>,
+) -> Result<TableIdent> {
+    // When querying a branch, spark sends something like the following as part of the `parameters`:
+    // namespace: (<my>, <namespace>, <table_name>)
+    // table_name: branch_<branch_name>
+
+    if let Some(identifier) = request_ident {
```

Comment:
can we somehow reduce nesting here and maybe clarify the table_name_candidate part a bit, perhaps with a comment?

Reply:
Do you have ideas?
I extended the comment.

Reply:
Something like this?:

```rust
fn determine_table_ident(
    parameters_ident: TableIdent,
    request_ident: Option<TableIdent>,
) -> Result<TableIdent> {
    // When querying a branch, spark sends something like the following as part of the `parameters`:
    // namespace: (<my>, <namespace>, <table_name>)
    // table_name: branch_<branch_name>
    let Some(identifier) = request_ident else {
        return Ok(parameters_ident);
    };

    if identifier == parameters_ident {
        return Ok(identifier);
    }

    let ns_parts = parameters_ident.namespace.clone().inner();
    let table_name_candidate = if ns_parts.len() >= 2 {
        NamespaceIdent::from_vec(ns_parts.iter().take(ns_parts.len() - 1).cloned().collect())
            .ok()
            .map(|n| TableIdent::new(n, ns_parts.last().cloned().unwrap_or_default()))
    } else {
        None
    };

    if table_name_candidate != Some(identifier.clone()) {
        return Err(ErrorModel::builder()
            .code(StatusCode::BAD_REQUEST.into())
            .message(
                "Table identifier in path does not match the one in the request body".to_string(),
            )
            .r#type("TableIdentifierMismatch".to_string())
            .build()
            .into());
    }

    Ok(identifier)
}
```

Reply:
Added